core/src/main/java/org/infinispan/protostream/impl/Log.java

```diff
@@ -151,6 +151,10 @@ default MalformedProtobufException messageTruncated() {
    @Message(value = "Incompatible @ProtoTypeId in '%s', from '%s' to '%s'", id = 40)
    String incompatibleTypeIds(String name, Integer t1, Integer t2);
 
+   @LogMessage(level = WARN)
+   @Message(value = "Unable to retrieve MethodHandles required for field based String optimisations. `--add-opens java.base/java.lang=ALL-UNNAMED` must be passed to the JVM", id = 41)
```

Comment:
Wondering if we want to word this a bit differently, maybe something like?

```suggestion
   @Message(value = "Unable to utilize more performant reflection based String serialization. Please pass `--add-opens java.base/java.lang=ALL-UNNAMED` to the JVM to enable", id = 41)
```